### ISSUE TITLE
Type error running on Flutter 3.3.10

### ISSUE THREAD

**Issue by TaylorHo** [RafaelBarbosatec/bonfire#372]
Hi, I spent some time without using Bonfire, and today I decided to use it again.

However, I'm getting an error that I still can't solve, and I would like your help!

### Given error:

```
flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
../../../.pub-cache/hosted/pub.dartlang.org/bonfire-2.11.11/lib/base/bonfire_game.dart:18:7: Error: The implementation of 'add' in the non-abstract class 'BonfireGame' does not conform to its
interface.
class BonfireGame extends BaseGame implements BonfireGameInterface {
      ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/flame-1.6.0/lib/src/components/core/component.dart:532:18: Context: The return type of the method 'Component.add' is 'FutureOr<void>', which does not
match the return type, 'Future<void>?', of the overridden method, 'BonfireGameInterface.add'.
 - 'Future' is from 'dart:async'.
Change to a subtype of 'Future<void>?'.
  FutureOr<void> add(Component component) => component.addToParent(this);
                 ^
../../../.pub-cache/hosted/pub.dartlang.org/bonfire-2.11.11/lib/base/bonfire_game_interface.dart:78:17: Context: This is the overridden method ('add').
  Future<void>? add(Component component);
                ^
../../../.pub-cache/hosted/pub.dartlang.org/bonfire-2.11.11/lib/game_interface/game_interface.dart:19:18: Error: A value of type 'FutureOr<void>' can't be returned from a function with return
type 'Future<void>?'.
 - 'Future' is from 'dart:async'.
    return super.add(component);
                 ^
Waiting for connection from debug service on Chrome...             13.9s
Failed to compile application.
```

### My Environment:

```
OS: macOS (arm M1)
Bonfire Version: 2.11.11
Flutter Version: 3.3.10
```

Thanks!

**Comment by RafaelBarbosatec** [comment]
This problem is occurring because of the new version of Flame. I will update soon.

**Comment by zhangyc** [comment]
me too

**Comment by RafaelBarbosatec** [comment]
If you use flame dependency in your project. try use the version 1.5.0, this is the problem.
I’m working in update flame to 1.6.0 in Bonfire.
This error is not about flutter version.

**Comment by RafaelBarbosatec** [comment]
Fixed in `2.12.0`
